### PR TITLE
Check gpg signature for library_index.json

### PR DESCRIPTION
**Pull request by cmaglie**
This PR implements the check for library_index.json.sig.
It has been implemented also the download of the gzipped version of the index to reduce bandwidth usage.

### CODE REVIEW

File: arduino/libraries/librariesmanager/download.go [hunk 1]
```diff
@@ -17,16 +17,13 @@ package librariesmanager
...
-
-	"go.bug.st/downloader/v2"
 )
 
-// LibraryIndexURL is the URL where to get library index.
+// LibraryIndexURL is the URL where to get the library index.
 var LibraryIndexURL, _ = url.Parse("https://downloads.arduino.cc/libraries/library_index.json")
 
-// UpdateIndex downloads the libraries index file from Arduino repository.
-func (lm *LibrariesManager) UpdateIndex(config *downloader.Config) (*downloader.Downloader, error) {
```
Comment (per1234) [inline arduino/libraries/librariesmanager/download.go:28]: Is this considered a breaking change?

Comment (cmaglie) [reply arduino/libraries/librariesmanager/download.go:28]: In theory, yes, but it was a so weird function that I doubt anyone has ever used it alone.
I'll add a note in the UPGRADING.md

Comment (per1234) [reply arduino/libraries/librariesmanager/download.go:28]: Haha, yeah I took a long dive down that rabbit hole when I was switching to use Arduino CLI's library index infrastructure in Arduino Lint (https://github.com/arduino/arduino-lint/commit/455001dfa1234abad1137389a22ab89c298c7cce), I ended up deciding to stick to the pre-existing bespoke code approach for the downloading after I got hopelessly lost, so Arduino Lint is not affected by this change.

I will take another look at what Arduino CLI has to offer if I get an opportunity to switch Arduino Lint to downloading the gzipped index.

File: docs/UPGRADING.md [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 Here you can find a list of migration guides to handle breaking changes between releases of the CLI.
 
+## 0.19.0
```
Comment (silvanocerza) [inline docs/UPGRADING.md:5]:
```suggestion
## Unreleased
```

The version will be set at release time.

**Review by silvanocerza — CHANGES_REQUESTED** [review]
Just a small doc thing.

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**